cmake/toolchain/aarch64-linux/crosscompile.cmake

```diff
@@ -1,6 +1,5 @@
 # CMake toolchain file for cross compiling to linux aarch64
 
-set(CROSS_COMPILE_ARM 1)
```

Comment:
This is used to build x265... I think we need to keep it (applies to both files with arm)

Reply:
How's that? All it did before was add `--arch=aarch64 --enable-cross-compile`, which is now added in a different way. It's not referenced anywhere else.

Reply:
https://github.com/LizardByte/build-deps/blob/4db40238879004780f18aa95330839434833978e/.github/workflows/build-ffmpeg.yml#L439-L451

That file is also used for AV1 and of course CBS.

Reply:
I should probably try to bring the whole thing into one CMake build

Reply:
Okay, I see what you mean now. In most cases, x265 checks for `ARM OR CROSS_COMPILE_ARM` with the former being based on `CMAKE_SYSTEM_PROCESSOR`. This would be true whether you were cross-compiling or not. The only difference I can see, at least for 64-bit, is that setting `CROSS_COMPILE_ARM` [leads to NEON being disabled](https://bitbucket.org/multicoreware/x265_git/src/3bd3dd731b4b4c3fbbe5e513c16bc6ae481a0ec5/source/CMakeLists.txt#lines-241:252), which presumably _isn't_ what you want. I don't know why they do that. In the master branch, they have added `CROSS_COMPILE_ARM64`, but this doesn't seem to make any difference at all.

Reply:
For what it's worth, Gentoo's x265 package can be cross-compiled even though it doesn't set those variables.

Reply:
I'm not sure if the resulting build would have any issues with sunshine when modifying the flags... but I can run it.

This probably won't be merged until after our next Sunshine release, just to be safe.

Reply:
Gentoo users are famous for modifying their flags. :laughing:

Reply:
Sorry, ignore the re-review, didn't realised it had failed since rebasing. I'll take a closer look at CBS.

Reply:
Okay, so on MacOS x86_64, CMake picks up Xcode by default. On aarch64, you explicitly tell it to use Clang. However, on both, ffmpeg previously chose gcc instead. Or perhaps gcc is just an alias for Clang here? In any case, I guess it's Clang you really want, so I'll probably need to make a toolchain file for x86_64.